timemachine/fe/absolute_hydration.py

```diff
@@ -155,3 +172,186 @@ def log_prob(xs, lam):
     resample = partial(smc.conditional_multinomial_resample, thresh=resample_thresh)
 
     return samples, lambdas, propagate, log_prob, resample
+
+
+def estimate_absolute_free_energy(
+    mol,
+    ff: Forcefield,
+    host_config: HostConfig,
+    seed: int,
+    n_frames=1000,
+    prefix="",
+    n_windows=None,
+    keep_idxs=None,
+    n_eq_steps=10000,
+    steps_per_frame=400,
+):
+    """
+    Estimate the absolute hydration free energy for the given mol.
+
+    Parameters
+    ----------
+    mol: Chem.Mol
+        molecule
+
+    ff: ff.Forcefield
+        Forcefield to be used for the system
+
+    host_config: HostConfig
+        Configuration for the host system.
+
+    n_frames: int
+        number of samples to generate for each lambda window, where each sample is `steps_per_frame` steps of MD.
+
+    prefix: str
+        A prefix to append to figures
+
+    seed: int
+        Random seed to use for the simulations.
+
+    n_windows: None
+        Number of windows used for interpolating the the lambda schedule with additional windows.
+
+    keep_idxs: list of int or None
+        If None, return only the end-state frames. Otherwise if not None, use only for debugging, and this
+        will return the frames corresponding to the idxs of interest.
+
+    n_eq_steps: int
+        Number of equilibration steps for each window.
+
+    steps_per_frame: int
+        The number of steps to take before collecting a frame
+
+    Returns
+    -------
+    SimulationResult
+        Collected data from the simulation (see class for storage information). We currently return frames
+        from only the first and last window.
+    """
+    bt = BaseTopology(mol, ff)
+    afe = AbsoluteFreeEnergy(mol, bt)
+
+    # note: tm convention lambda=1 means "decoupled", lambda=0 means "coupled"
+    lambda_schedule = construct_pre_optimized_absolute_lambda_schedule_solvent(n_windows)[::-1]
+    assert np.isclose(lambda_schedule[0], 1.0) and np.isclose(lambda_schedule[-1], 0.0)
```

Comment:
nit: any reason not to assert equality here?

Reply:
just used the same code as above